frontend/www/js/omegaup/problem/creator/modules/cases.ts

```diff
@@ -226,7 +227,119 @@ export const casesStore: Module<CasesState, RootState> = {
+      let caseNumber = 0;
+      for (let i = 0; i < multipleCaseRequest.numberOfCases; i++) {
+        let shouldBreak = false;
+        do {
+          caseNumber++;
+          const name =
+            multipleCaseRequest.prefix +
+            caseNumber +
+            multipleCaseRequest.suffix;
+          if (multipleCaseRequest.groupID === UUID_NIL) {
+            // Should search in all groups
+            const groupExist = state.groups.find(
+              (group) => group.name === name,
+            );
+            if (groupExist) {
+              continue;
+            }
+          } else {
+            // Should search in the given group
+            const caseExist = cases.find((_case) => _case.name === name);
+            if (caseExist) {
+              continue;
+            }
+          }
+          const newCase = generateCase({
+            name: name,
+            groupID: multipleCaseRequest.groupID,
+          });
+          const caseRequest = generateCaseRequest({
+            ...newCase,
+            points: 0,
+            pointsDefined: false,
+          });
+          commit('addCase', caseRequest);
+          shouldBreak = true;
+        } while (!shouldBreak);
+      }
+    },
+    setLines({ getters }, lines: CaseLine[]) {
+      const selectedCase: Case = getters.getSelectedCase;
+      selectedCase.lines = lines;
+    },
+    addNewLine({ getters }) {
+      const selectedCase: Case = getters.getSelectedCase;
+      const newLine: CaseLine = {
+        lineID: uuid(),
+        label: 'NEW',
+        data: {
+          kind: 'line',
+          value: '',
+        },
+      };
+      selectedCase.lines.push(newLine);
+    },
+    updateLine({ getters }, newLine: CaseLine) {
+      const selectedCase: Case = getters.getSelectedCase;
+      const lineToUpdate = selectedCase.lines.find(
+        (line) => line.lineID === newLine.lineID,
+      );
+      if (lineToUpdate === undefined) {
+        return;
+      }
+      lineToUpdate.label = newLine.label;
+      lineToUpdate.data = newLine.data;
+    },
+    deleteLine({ getters }, lineIDToBeDeleted: LineID) {
+      const selectedCase: Case = getters.getSelectedCase;
+      selectedCase.lines = selectedCase.lines.filter(
+        (line) => line.lineID !== lineIDToBeDeleted,
+      );
+    },
+  },
+  getters: {
+    getCasesFromGroup: (state) => (groupID: GroupID) => {
+      return state.groups.find((group) => group.groupID === groupID)?.cases;
+    },
+    getGroupIdsAndNames: (state) => {
+      return state.groups.map((group) => {
+        return { value: group.groupID, text: group.name };
+      });
+    },
+    getAllCases: (state) => {
+      return state.groups.reduce((cases: Case[], currCase) => {
+        return [...cases, ...currCase.cases];
+      }, []);
+    },
+    getSelectedCase: (state) => {
+      const selectedGroup = state.groups.find(
+        (group) => group.groupID === state.selected.groupID,
+      );
+      if (selectedGroup !== undefined) {
```

Comment:
aquí podemos usar el [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html), donde los `if`s denotan "errores" y al final de la cadena es el caminito feliz donde todo salió bien:

```suggestion
      if (selectedGroup === undefined) {  // o !selectedGroup para ser más concisos.
          return null;  // null es mejor que undefined.
      }
      // L331-333, pero con `?? null;` para regresar `null`.
```